### DISCUSSION TITLE
How to Flash Firmware

### DISCUSSION THREAD

**Discussion by ic-racer** [srozum/film_camera_tester#21] [Q&A] [answered]
I'm preparing to build the device and am looking at the firmware requirements. 
I have no experience with .cpp files. I'm using macintosh, what hardware and software would I need? Do I need two Arduino boards to do the flash? Any special cables? 

I know the kits are gone, but could I get an Arduino already flashed to simplify this step?

Thanks, IC

**Comment by srozum** [comment, marked as answer]
Hello,

I'll recommend to read Wiki first, it also has a page about how to upload the firmware https://github.com/srozum/film_camera_tester/wiki/Firmware

Avrdude utility is used to program Arduino. It usually comes with Arduino IDE or Platform IO extension, and can be installed separately using brew on mac.

**Reply by ic-racer** [reply]
Thank you, my understanding is that I either need to buy a USBasp programmer or wire two Arduino boards together. 

Can I use this USBasp programmer in the link here :  https://www.smart-prototyping.com/USBASP-AVR-Programmer.html

I presume it will be self-explanitory where the USBasp's plug on the ribbon cable will plug into my Arduino board?

The second problem, running avrdude, I believe is solved in the OSX terminal program as follows:

> /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" 
> echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> ~/.zprofile
> brew install avrdude

Are you sure I can't just pay you for a flashed board?

Thank, you.

**Reply by srozum** [reply]
> my understanding is that I either need to buy a USBasp programmer or wire two Arduino boards together.

correct. I would suggest to get 3 boards so you have spare just in case, and use one as a programmer

> Can I use this USBasp programmer in the link here

yes

> The second problem, running avrdude

that command should work. Easier would be to install Arduino IDE and then find out where it installed avrdude. With IDE you also can try and test your boards.

> Are you sure I can't just pay you for a flashed board?

I used to have extra boards on a Tindie store, but all sold out. Sorry.
